tests/test_rp_exec.py

```diff
@@ -50,97 +51,165 @@ def get_rp_decorator():
-    # TODO: Figure out what `activate` does that `rp-venv/bin/python` doesn't do.
     with rp.Session() as session:
-        # Based on `radical.pilot/examples/config.json`
-        # TODO: Does the Session have a default spec for 'local.localhost'? Can/should we reference it?
-        # See also https://github.com/radical-cybertools/radical.pilot/issues/2181
-        resource = 'local.localhost'
-        resource_config = {resource: {}}
-        if resource in rp_config['config']:
-            resource_config[resource].update(rp_config.config[resource])
-        resource_config[resource].update({
-            'project': None,
-            'queue': None,
-            'schema': None,
-            'cores': 1,
-            'gpus': 0
-        })
-
-        pilot_description = dict(resource=resource,
-                                 runtime=30,
-                                 exit_on_error=True,
-                                 project=resource_config[resource]['project'],
-                                 queue=resource_config[resource]['queue'],
-                                 cores=resource_config[resource]['cores'],
-                                 gpus=resource_config[resource]['gpus'])
-
-        task_description = {'executable': '/bin/date',
-                            'cpu_processes': 1,
-                            }
-
-        pmgr = rp.PilotManager(session=session)
-        umgr = rp.UnitManager(session=session)
-        pilot = pmgr.submit_pilots(rp.ComputePilotDescription(pilot_description))
-        task = umgr.submit_units(rp.ComputeUnitDescription(task_description))
-
-        umgr.add_pilots(pilot)
-        umgr.wait_units()
 
-        assert task.exit_code == 0
+        # define a pilot and launch it
+        pd    = rp.PilotDescription(
+                   {'resource': 'local.localhost',
+                    'cores'   : 4,
+                    'gpus'    : 0})
+
+        pmgr  = rp.PilotManager(session=session)
+        tmgr  = rp.TaskManager(session=session)
+        pilot = pmgr.submit_pilots(pd)
+        tmgr.add_pilots(pilot)
+
+        # TODO: How can we recover successful components of previous failed Sessions?
+        #    Need to reference sandboxes from other sessions (RCT work in progrees).
+
+        # define a raptor.scalems master and launch it within the pilot
+        pwd   = os.path.dirname(__file__)
+        td    = rp.TaskDescription(
+                {
+                    'uid'          :  'raptor.scalems',
+                    'executable'   :  'python3',
+                    'arguments'    : ['./scalems_test_master.py', '%s/scalems_test_cfg.json'  % pwd],
+                    'input_staging': ['%s/scalems_test_cfg.json'  % pwd,
+                                      '%s/scalems_test_master.py' % pwd,
+                                      '%s/scalems_test_worker.py' % pwd]
+                })
+        scheduler = tmgr.submit_tasks(td)
+
+        # FIXME
```

Comment:
This should not be needed anymore

Reply:
Thanks. I will remove in an additional commit.